dapps/profile.vue/src/components/did/DID.ts

```diff
@@ -19,49 +19,145 @@
 
 import Component, { mixins } from 'vue-class-component';
 import { EvanComponent } from '@evan.network/ui-vue-core';
-import { Runtime } from '@evan.network/api-blockchain-core';
-import { Delegate, DidDocumentTemplate, ServiceEndpoint } from './DidInterfaces';
+import { Runtime, DidDocument } from '@evan.network/api-blockchain-core';
+import { isEqual } from 'lodash';
+import { Delegate, ServiceEndpoint } from './DidInterfaces';
 import { DidService } from './DidService';
 
 @Component
 export default class DIDComponent extends mixins(EvanComponent) {
-  didDocument: DidDocumentTemplate = null;
+  didDocument: DidDocument = null;
 
   delegates: Delegate[] = null;
 
   endpoints: ServiceEndpoint[] = null;
 
+  isEditMode = false;
+
+  isLoading = false;
+
+  previousDelegates: Delegate[] = null;
+
+  previousEndpoints: ServiceEndpoint[] = null;
+
   runtime: Runtime = null;
 
+  didService: DidService;
+
   onPageEntries = [
     {
       id: 'did',
       label: this.$t('_profile.did.did-title'),
     },
     {
       id: 'service-endpoints',
-      label: this.$t('_profile.did.service-endpoint-title'),
+      label: this.$t('_profile.did.service-endpoints-title'),
     },
     {
       id: 'delegates',
       label: this.$t('_profile.did.delegates-title'),
     },
   ]
 
+  get hasEditRights(): boolean {
+    // Check if owner
+    if (this.runtime.activeIdentity === this.didDocument.id) {
+      return true;
+    }
+    // Check if current user is controller
+    if (this.didService.getControllers(this.didDocument).indexOf(this.runtime.activeIdentity)) {
+      return true;
+    }
+    return false;
+  }
+
   async created(): Promise<void> {
-    this.didDocument = await DidService.fetchDidDocument();
-    this.delegates = await DidService.getDelegates(this.didDocument);
-    this.endpoints = await DidService.getServiceEndpoints(this.didDocument);
-    console.log('this.didDocument', this.didDocument);
-    console.log('this.delegates', this.delegates);
-    /* // TODO: switch after complete identity switch to: runtime.did.getDidDocument();
-       const identity = await this.runtime.verifications.getIdentityForAccount(this.runtime.activeAccount, true);
-       const did = await this.runtime.did.convertIdentityToDid(identity);
-       const document = await this.runtime.did.getDidDocumentTemplate();
-       await this.runtime.did.setDidDocument(did, document);
-       const retrieved = await this.runtime.did.getDidDocument(did); */
+    this.didService = new DidService(this.getRuntime());
```

Comment:
Please check at the beginning, if a new identity based profile is loaded. If not, show a message, that the did functionality is not supported with old profiles.